modules/imgcodecs/src/loadsave.cpp

```diff
@@ -714,6 +715,19 @@ static bool imwrite_( const String& filename, const std::vector<Mat>& img_vec,
     bool code = false;
     try
     {
+        FILE* f = fopen( filename.c_str(), "wb" );
+        if (f == 0)
+        {
+            if (errno == EACCES)
+            {
+                CV_Error( Error::StsError, "permission denied" );
```

Comment:
Please use logging:
```
CV_LOG_WARNING(NULL, "imwrite(): Can't open file for writing: " << filename);
return false;
```

Reply:
I have updated my code accordingly. I did initially consider trying for file access in all encoders however, as you've said, the implementations of encoder.write are different. This could potentially raise more issues in the process and may not be so elegant. Thus I've opted for second suggestion.